chore: make command timeout parameter explicit

crenshaw-dev opened the pull request:
This is a very large PR to do a very small thing - instead of loading a global command timeout from an env var in a util module, load it explicitly from env vars or CLI flags in each place that uses the command util module and pass it explicitly.

Why? Because 
1) The env var is documented in only one place, and it's not at all clear that it's used in _tons_ of places
2) The code doesn't currently make clear that external calls and a timeout are involved - this change makes the code's behavior clearer
3) In many places, this timeout should probably be replaced by a context. For example, in calls to the repo-server, we should probably just run commands with the gRPC-determined timeout rather than tacking on another one. This PR moves us towards that eventual refactor.

crenshaw-dev commented on util/exec/exec.go:
```diff
@@ -12,25 +11,11 @@ import (
 	"github.com/argoproj/argo-cd/v2/util/log"
 )
 
-var timeout time.Duration
-
-func init() {
-	initTimeout()
-}
-
-func initTimeout() {
-	var err error
-	timeout, err = time.ParseDuration(os.Getenv("ARGOCD_EXEC_TIMEOUT"))
-	if err != nil {
-		timeout = 90 * time.Second
-	}
-}
```
This is what I wanted to get rid of.

crenshaw-dev commented:
@jannfis agreed! I thought I had the `ARGOCD_EXEC_TIMEOUT` reverse-compatibility included everywhere, but I see that I did not. I'll add that.

jannfis commented:
And sorry, I have neither read your your comment nor the code thoroughly :( Too long of a day today.

Obviously, you mentioned that you take the env var from the command line flags, which is good enough.

crenshaw-dev commented:
You were right, one of the components was missing a fallback.

I've added the missing fallback, consolidated the fallback logic in a util function, tested the util function, and added a deprecation notice to the release notes.

todaywasawesome commented:
@crenshaw-dev Can you fix the DCO on these commits? Not a huge deal.

pasha-codefresh commented:
@crenshaw-dev do you think we still will process with this PR? i can try to find time for review, just let me know if you still want to contribute this feature

ishitasequeira reviewed (commented):
@crenshaw-dev are we still moving forward with this PR?

andrii-korotkov-verkada commented:
Closed the PR due to it being old and unlikely to be able to resolve merge conflicts. But feel free to re-open/re-submit if the work is still planned.
